library/modules/Gui.cpp

```diff
@@ -1111,6 +1112,26 @@ df::building_stockpilest* Gui::getSelectedStockpile(color_ostream& out, bool qui
     return stockpile;
 }
 
+bool Gui::any_civzone_hotkey(df::viewscreen* top) {
+    return getAnyCivZone(top) != NULL;
+}
+
+df::building_civzonest *Gui::getAnyCivZone(df::viewscreen* top) {
+    if (matchFocusString("dwarfmode/Zone")) {
+        return game->main_interface.civzone.cur_bld;
+    }
```

Comment:
Just to check, does this get the right value when multiple zones overlap?

Reply:
I shall check tonight, I more or less copy and pasted this logic from the other `get` functions.

Reply:
ack. the reason I'm suspicious is that civzones are the only building type that can overlap, so the rules might be different.

Reply:
Just tested, this works properly when they're completely overlapped (tested a meeting hall and an office stacked on top of each other exactly, when I change the selection using the left and right arrows here:

![image](https://user-images.githubusercontent.com/5893/218343969-eed9994c-a2ca-4d3a-bb3e-187e0badccb0.png)

the function returns the correct civzone.